### PR TITLE
Add auto formatting for doxygen start tag to prepare script

### PR DESCRIPTION
**Pull request by nyalldawson**
Neateans doxygen comments by changing:

```
/** my thing
  */
```

to

```
/**
 * my thing
 */
```

and

```
/**
  * some thing
  */
thing
/**
 * another thing
  */
```

to 

```
/**
  * some thing
  */
thing

/**
 * another thing
  */
```

and 

```
SomeValue, /*!< my enum val */ 
```

to

```
SomeValue, //!< My enum val
```

### CODE REVIEW

File: scripts/doxygen_space.pl [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/perl -0 -i.sortinc -n
+###########################################################################
+#    unify_includes.pl
```
Comment (jef-n) [inline scripts/doxygen_space.pl]: wrong header

File: src/analysis/interpolation/Bezier3D.h [hunk 2]
```diff
@@ -20,45 +20,76 @@
...
  * Class Bezier3D represents a bezier curve, represented by control points. Parameter t is running from 0 to 1. The class is capable to calculate the curve point and the first two derivatives belonging to t.*/
 class ANALYSIS_EXPORT Bezier3D: public ParametricLine
 {
   protected:
 
   public:
-    /** Default constructor*/
+
+    /**
+     * Default constructor*/
```
Comment (jef-n) [inline src/analysis/interpolation/Bezier3D.h]: IMHO a wast of vertical space.  One line comments should stay on one line. OTOH if we put `/**` on a separate line the `*/` should also be on a separate line.

File: src/analysis/interpolation/CloughTocherInterpolator.h [hunk 3]
```diff
@@ -22,20 +22,31 @@
 
 class NormVecDecorator;
 
-/** \ingroup analysis
+/**
+ * \ingroup analysis
  * This is an implementation of a Clough-Tocher interpolator based on a triangular tessellation. The derivatives orthogonal to the boundary curves are interpolated linearly along a triangle edge.*/
```
Comment (jef-n) [inline src/analysis/interpolation/CloughTocherInterpolator.h]: side note: word wrapping could be interesting too

File: src/analysis/interpolation/qgsidwinterpolator.h [hunk 4]
```diff
@@ -29,7 +30,8 @@ class ANALYSIS_EXPORT QgsIDWInterpolator: public QgsInterpolator
     QgsIDWInterpolator( const QList<LayerData>& layerData );
     ~QgsIDWInterpolator();
 
-    /** Calculates interpolation value for map coordinates x, y
+    /**
+     * Calculates interpolation value for map coordinates x, y
```
Comment (jef-n) [inline src/analysis/interpolation/qgsidwinterpolator.h]: doesn't add the rest of the `*`

File: src/analysis/raster/qgsrastercalculator.h [hunk 5]
```diff
@@ -44,15 +45,16 @@ class ANALYSIS_EXPORT QgsRasterCalculator
     //! Result of the calculation
     enum Result
     {
-      Success = 0, /*!< Calculation sucessful */
-      CreateOutputError = 1, /*!< Error creating output data file */
-      InputLayerError = 2, /*!< Error reading input layer */
-      Cancelled = 3, /*!< User cancelled calculation */
-      ParserError = 4, /*!< Error parsing formula */
-      MemoryError = 5, /*!< Error allocating memory for result */
+      Success = 0, //!< Calculation sucessful
```
Comment (jef-n) [inline src/analysis/raster/qgsrastercalculator.h]: side note: typo

File: src/analysis/raster/qgsrastercalculator.h [hunk 6]
```diff
@@ -78,7 +81,8 @@ class ANALYSIS_EXPORT QgsRasterCalculator
     QgsRasterCalculator( const QString& formulaString, const QString& outputFile, const QString& outputFormat,
                          const QgsRectangle& outputExtent, const QgsCoordinateReferenceSystem& outputCrs, int nOutputColumns, int nOutputRows, const QVector<QgsRasterCalculatorEntry>& rasterEntries );
 
-    /** Starts the calculation and writes new raster
+    /**
+     * Starts the calculation and writes new raster
```
Comment (jef-n) [inline src/analysis/raster/qgsrastercalculator.h]: more occasions where a leading `*` is added only to the first line of a multiline comment.

File: src/app/qgsmaptoolsimplify.h [hunk 7]
```diff
@@ -71,12 +72,15 @@ class APP_EXPORT QgsMapToolSimplify: public QgsMapToolEdit
...
+
+    /**
+     * Slot to change display when slidebar is moved */
     void setTolerance( double tolerance );
 
     void setToleranceUnits( int units );
 
-    /** Slot to store feture after simplification */
+    /**
+     * Slot to store feture after simplification */
```
Comment (jef-n) [inline src/app/qgsmaptoolsimplify.h]: side note: typo

**Review by jef-n — COMMENTED** [review]
Undecided.  A lot of changes with room of improvement and IMHO little reward.

File: src/analysis/interpolation/Bezier3D.h (hunk 2 above)
Comment (nyalldawson) [inline src/analysis/interpolation/Bezier3D.h]: Maybe we should auto-flip these to a single line `//! ` tag?

I realised the trailing */ issue too. That definitely needs addressing.

File: src/analysis/raster/qgsrastercalculator.h (hunk 5 above)
Comment (nyalldawson) [inline src/analysis/raster/qgsrastercalculator.h]: Fixed in https://github.com/qgis/QGIS/commit/02ea2d77036e929c478600172e564727598f11f3

File: src/app/qgsmaptoolsimplify.h (hunk 7 above)
Comment (nyalldawson) [inline src/app/qgsmaptoolsimplify.h]: Fixed in https://github.com/qgis/QGIS/commit/02ea2d77036e929c478600172e564727598f11f3

**Comment by nyalldawson** [comment]
Ok, the useful parts of this have all been manually merged. Outstanding parts are:

- Handling:
 /** my thing
  */

-  word wrapping for too long documentation lines

**Pull request closed without merging**